flang/lib/Optimizer/Transforms/RaiseToAffine.cpp

```diff
@@ -9,40 +9,116 @@
 #include "PassDetail.h"
 #include "flang/Optimizer/Dialect/FIRDialect.h"
 #include "flang/Optimizer/Dialect/FIROps.h"
+#include "flang/Optimizer/Transforms/AffinePromotion.h"
 #include "flang/Optimizer/Transforms/Passes.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Visitors.h"
 #include "mlir/Dialect/Affine/IR/AffineOps.h"
 #include "mlir/Dialect/StandardOps/IR/Ops.h"
 #include "mlir/Pass/Pass.h"
 #include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/DenseMap.h"
 #include "llvm/Support/CommandLine.h"
+#define DEBUG_TYPE "flang-affine-promotion"
+#include <iostream>
 
 /// disable FIR to affine dialect conversion
 static llvm::cl::opt<bool>
     disableAffinePromo("disable-affine-promotion",
                        llvm::cl::desc("disable FIR to Affine pass"),
                        llvm::cl::init(false));
 
-using namespace fir;
+namespace fir {
```

Comment:
Another LLVM thing. Use `fir::AffineFunctionAnalysis::AffineFunctionAnalysis` and omit the namespace block.

Reply:
Should I even keep these classes in fir namespace? Looking at other files in the directory most of the heavy lifting is done under anonymous namespace.

Reply:
True. If the code is only going to be used in affine promotion then it probably doesn't need to be in the public interface. Good point.

Reply:
I had a recursive dependency earlier but that was not a good design, I moved them all to cpp now.